python/dgl/distgnn/partition/libra_partition.py

```diff
@@ -0,0 +1,577 @@
+"""
+Copyright (c) 2021 Intel Corporation
+ \file distgnn/partition/libra_partition.py
+ \brief Libra - Vertex-cut based graph partitioner for distirbuted training
+ \author Vasimuddin Md <vasimuddin.md@intel.com>,
+         Guixiang Ma <guixiang.ma@intel.com>
+         Sanchit Misra <sanchit.misra@intel.com>,
+         Ramanarayan Mohanty <ramanarayan.mohanty@intel.com>,         
+         Sasikanth Avancha <sasikanth.avancha@intel.com>
+         Nesreen K. Ahmed <nesreen.k.ahmed@intel.com>
+ \cite Distributed Power-law Graph Computing: Theoretical and Empirical Analysis
+"""
+
+import os
+import sys
+import networkx as nx
+import numpy as np
+import csv
+from statistics import mean
+import random
+import time
+
+import dgl
+import json
+from dgl import DGLGraph
+import torch as th
+from dgl.data import register_data_args, load_data
+from load_graph import load_reddit, load_ogb
+from dgl.sparse import libra_vertex_cut
+from dgl.sparse import libra2dgl_build_dict
+from dgl.sparse import libra2dgl_set_lf
+from dgl.sparse import libra2dgl_build_adjlist
+from dgl.data.utils import load_graphs, save_graphs, load_tensors, save_tensors
+from scipy.io import mmread
+from dgl.base import DGLError
+import requests
+
+## \brief This function reports replication per node
+def rep_per_node(prefix, nc):
+    ifile = os.path.join(prefix, 'replicationlist.csv')
+    f = open(ifile, "r")
+    r_dt = {}
+
+    fline = f.readline()
+    for line in f:
+        if line[0] == '#':
+            raise DGLError("Error: Read hash in rep_per_node func.")
+    
+        node = line.strip('\n')
+        if r_dt.get(node, -100) == -100:
+            r_dt[node] = 1
+        else:
+            r_dt[node] += 1
+
+    f.close()
+    ## checks
+    for v in r_dt.values():
+        if v >= nc:
+            raise DGLError("Error: Unexpected event in rep_per_node func.")
+
+    return r_dt
+
+
+## \brief This function converts libra output to dgl format
+def main_libra2dgl(resultdir, dataset, nc):
+    """
+    Converts the output from Libra partitioning to DGL/DistGNN graph input.
+    It builds dictionaries to assign local IDs to nodes in the partitions as well
+    as it build a database to keep track of the location of clone nodes in the remote
+    partitions.
+
+    Parameters
+    ----------
+    resultdir : Location where partitions in dgl format are stored
+    dataset : Dataset name
+    nc : Number of partitions
+
+    Output
+    ------
+    Creates partX folder in resultdir location for each partition X
+
+    Notes
+    -----
+    This output is directly used as input to DistGNN
+    
+    """
+    tedges = 1615685872   ## total edges
+    max_c = 1024   ## max partitions supported
+    factor = 1.2
```

Comment:
What are these numbers?

Reply:
Removed these numbers - no longer required. These were the calculations for the size of memory pre-allocation for libra output to DGL input routine. Now, the memory pre-allocation size is provided by the libra_vertex_cut routine.